src/main/java/org/jenkinsci/plugins/github/webhook/GHEventsListener.java

```diff
@@ -0,0 +1,76 @@
+package org.jenkinsci.plugins.github.webhook;
+
+import com.google.common.annotations.Beta;
+import com.google.common.base.Function;
+import com.google.common.base.Predicate;
+import hudson.ExtensionList;
+import hudson.ExtensionPoint;
+import hudson.model.AbstractProject;
+import jenkins.model.Jenkins;
+import org.kohsuke.github.GHEvent;
+
+import java.util.Set;
+
+/**
+ * Extension point to contribute events plugin interested in.
+ * This point should return true in {@link #isApplicable(AbstractProject)}
+ * only if it can parse hooks with events contributed in {@link #events()}
+ *
+ * Each time this plugin wants to get events list from contributors it asks for applicable status
+ *
+ * @author lanwen (Merkushev Kirill)
+ * @since TODO
+ */
+public abstract class GHEventsListener implements ExtensionPoint {
```

Comment:
This extension point slightly differs from common Jenkins ones. Listener is supposed to get events somehow. I think it's "Subscriber", not "Listener"